File: 3rdDay/palindrome.c
#include<stdio.h>
int palindrome(char *str,int low, int high){
    if(low>=high){
        return 1;
    }
    if(str[low]!=str[high]){
        return 0;
    }
    palindrome(str,low+1,high-1);
    // return 1;
}
int main(){
    char *str;
    scanf("%s",str);
    int size= (sizeof(str)/sizeof(str-1))-1;
    int low=0;
    int high=size-1;
    if(palindrome(str,low,high)){
        printf("string is palindrome");

    }
    else{
        printf("string is not palindrome");
    }

}

File: 9thDay/palindrome.java
import java.util.*;

public class palindrome {

    public static Boolean reverse(String str){
        int low=0;
        int high=str.length()-1;
        while(low<=high){
            if(str.charAt(low)!=str.charAt(high-low)){
                return false;
            }
            low++;
        }
        return true;
        }
    
    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        String str = sc.nextLine();
        boolean x = reverse(str);
        if(x==true){
            System.out.println("The string is a palindrome");
        }
        else{
            System.out.println("The string is not a palindrome");
        }
    }
}
